tests/test_stumped.py

```diff
@@ -150,3 +150,87 @@ def test_stumped_A_B_join_df(T_A, T_B, dask_client):
     utils.replace_inf(left)
     utils.replace_inf(right)
     npt.assert_almost_equal(left, right)
+
+
+def test_stumped_nan_self_join_beginning(dask_client):
+    m = 3
+    T = np.array([np.nan, 1, 0, 0, 1, 0, 0])
+
+    zone = int(np.ceil(m / 4))
+    left = np.array(
+        [
+            utils.naive_mass(Q, T, m, i, zone, ignore_trivial=True)
+            for i, Q in enumerate(core.rolling_window(T, m))
+        ],
+        dtype=object,
+    )
+
+    right = stumped(dask_client, T, m, ignore_trivial=True)
+
+    utils.replace_inf(left)
+    utils.replace_inf(right)
+    npt.assert_almost_equal(left[:, 0], right[:, 0])
+
+
+def test_stumped_inf_self_join_beginning(dask_client):
+    m = 3
+    T = np.array([np.inf, 1, 0, 0, 1, 0, 0])
+
+    zone = int(np.ceil(m / 4))
+    left = np.array(
+        [
+            utils.naive_mass(Q, T, m, i, zone, ignore_trivial=True)
+            for i, Q in enumerate(core.rolling_window(T, m))
+        ],
+        dtype=object,
+    )
+
+    right = stumped(dask_client, T, m, ignore_trivial=True)
+
+    utils.replace_inf(left)
+    utils.replace_inf(right)
+    npt.assert_almost_equal(left[:, 0], right[:, 0])
+
+
+@pytest.mark.parametrize("T_A, T_B", test_data)
+def test_stumped_nan_inf_self_join(T_A, T_B, dask_client):
+    m = 3
+    nan_inf_sample_B = np.random.permutation(np.arange(len(T_B)))
+    T_B[nan_inf_sample_B[: len(T_B) // 20]] = np.nan
+    T_B[nan_inf_sample_B[-len(T_B) // 20 :]] = np.inf
+
+    zone = int(np.ceil(m / 4))
+    left = np.array(
+        [
+            utils.naive_mass(Q, T_B, m, i, zone, ignore_trivial=True)
+            for i, Q in enumerate(core.rolling_window(T_B, m))
+        ],
+        dtype=object,
+    )
+
+    right = stumped(dask_client, T_B, m, ignore_trivial=True)
+
+    utils.replace_inf(left)
+    utils.replace_inf(right)
+    npt.assert_almost_equal(left, right)
+
+
+@pytest.mark.parametrize("T_A, T_B", test_data)
+def test_stumped_nan_inf_A_B_join(T_A, T_B, dask_client):
+    m = 3
+    nan_inf_sample_A = np.random.permutation(np.arange(len(T_A)))
```

Comment:
This seems overly complicated for a unit test. It is more important that these particular test cases have hard coded `np.nan` and `np.inf` positions.